openpype/modules/shotgrid/lib/settings.py

```diff
@@ -0,0 +1,39 @@
+import os
+from typing import Tuple, Dict, List, Any
+
+from pymongo import MongoClient
+from openpype.api import get_system_settings, get_project_settings
+from openpype.modules.shotgrid.lib.const import MODULE_NAME
+
+
+def get_project_list() -> List[str]:
+    mongo_url = os.getenv("OPENPYPE_MONGO")
+    client = MongoClient(mongo_url)
+    db = client['avalon']
+    return db.list_collection_names()
+
+
+def get_shotgrid_project_settings(project: str) -> Dict[str, Any]:
+    return get_project_settings(project).get(MODULE_NAME, {})
+
+
+def get_shotgrid_settings() -> Dict[str, Any]:
+    return get_system_settings().get("modules", {}).get(MODULE_NAME, {})
+
+
+def get_shotgrid_servers() -> Dict[str, Any]:
+    return get_shotgrid_settings().get("shotgrid_settings", {})
+
+
+def get_leecher_backend_url() -> str:
+    return get_shotgrid_settings().get("leecher_backend_url")
+
+
+def filter_projects_by_login() -> bool:
+    return bool(get_shotgrid_settings().get("filter_projects_by_login", False))
```

Comment:
These functions won't be very optimized - it might make sense more to instead add the result of `get_shotgrid_settings` as an argument. That way you don't need constantly requery the database for the settings (which can be quite a lot of data to query).

It's not too many calls currently - but it'd remove a call [here](https://github.com/pypeclub/OpenPype/pull/2921/files#diff-e649a47dc6911d202e5f2ca3ccae4a0c20bd81cb81a10e03e60306bac27db8cdR44).

Thoughts?

Reply:
> These functions won't be very optimized

_Premature optimization is the root of all evil (or at least most of it) in programming_ - [Donald Knuth](https://en.wikipedia.org/wiki/Donald_Knuth)

We are talking about usage of MongoDB, which is reputed of being highly performant and has strongly optimized throughput. 
Despite the fact that it's a premature optimization. I've added LRU cache for those functions.